pkg/kubelet/kuberuntime/kuberuntime_sandbox_linux.go

```diff
@@ -51,7 +51,19 @@ func (m *kubeGenericRuntimeManager) calculateSandboxResources(pod *v1.Pod) *runt
 	if _, cpuRequestExists := req[v1.ResourceCPU]; cpuRequestExists {
 		cpuRequest = req.Cpu()
 	}
-	return m.calculateLinuxResources(cpuRequest, lim.Cpu(), lim.Memory())
+
+	lcr := m.calculateLinuxResources(cpuRequest, lim.Cpu(), lim.Memory())
+	// cfs quota for container is not capped if
+	// 1. cpu manager policy is static
+	// 2. pod has quos PodQOSGuaranteed
+	// 3. container has integer cpu request
+	if m.containerManager.cpuExperimentalManagerPolicyStatic &&
+		kubeapiqos.GetPodQOS(pod) == v1.PodQOSGuaranteed &&
+		cpuRequest.MilliValue() % MilliCPUToCPU == 0 {
```

Comment:
I believe this is still disabling quota even when ContainerHasIntCpuReuquest is false.

Reply:
last item in that if chain `cpuRequest.MilliValue() % MilliCPUToCPU == 0 ` should prevent that

Reply:
let's move the logic in the cpumanager, either with the current approach or checking if the container had exclusive cpu assignments. If we keep this logic, which I'm not against, let's add unit test coverage.

Reply:
done! will work on adding unit test coverage. let me know if we want to check precision of cpu such that "1" is pinned but "1.0" is not